arch/sim/src/sim/CMakeLists.txt

```diff
@@ -174,6 +174,9 @@ endif()
 
 if(CONFIG_SMP)
   list(APPEND SRCS sim_smpsignal.c sim_cpuidlestack.c)
+endif()
+
+if(CONFIG_ARCH_HAVE_MULTICPU)
```

Comment:
@hujun260 only POSIX implementation is provided here, no Win implementation 
this will block MSVC build

Reply:
let's check windows before select ARCH_HAVE_MULTICPU, @hujun260

Reply:
ok